### PR TITLE
#4885 Add the missing stubs in stdlib/tarfile

### PR DESCRIPTION
**Pull request by asing1**

### CODE REVIEW

File: stdlib/tarfile.pyi [hunk 1]
```diff
@@ -81,7 +86,8 @@ class TarFile(Iterable[TarInfo]):
     dereference: Optional[bool]
     ignore_zeros: Optional[bool]
     encoding: Optional[str]
-    errors: str
+    errors: Optional[str]
```
Comment (hauntsaninja) [inline stdlib/tarfile.pyi:89]: I think this should still be `errors: str`. I know the class level declaration has `errors = None`, but instances will end up having `errors` as a `str` and I believe it gets passed to various functions that expect str.
```suggestion
    errors: str
```

**Review by hauntsaninja — CHANGES_REQUESTED** [review]
Thanks, this looks great! One suggestion.

File: stdlib/tarfile.pyi (hunk 1 above)
Comment (asing1) [reply stdlib/tarfile.pyi]: Removed the optional part from type hinting as mentioned

**Comment by hauntsaninja** [comment]
Thank you! :-)

**Pull request merged**